cli/cmd/lql_delete.go

```diff
@@ -0,0 +1,76 @@
+//
+// Author:: Salim Afiune Maya (<afiune@lacework.net>)
+// Copyright:: Copyright 2020, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/lacework/go-sdk/api"
+)
+
+const (
+	lqlDeleteDebugMsg   string = "deleting LQL query"
+	lqlDeleteSuccessMsg string = "LQL query (%s) deleted successfully.\n"
+	lqlDeleteUnableMsg  string = "unable to delete LQL query"
+)
+
+var (
+	// lqlDeleteCmd represents the lql delete command
+	lqlDeleteCmd = &cobra.Command{
+		Use:   "delete <query_id>",
+		Short: "delete an LQL query",
+		Long:  `Delete an LQL query.`,
+		Args:  cobra.MaximumNArgs(1),
+		RunE:  deleteQuery,
+	}
+)
+
+func init() {
+	// add sub-commands to the lql command
+	lqlCmd.AddCommand(lqlDeleteCmd)
+}
+
+func deleteQuery(_ *cobra.Command, args []string) error {
+	var queryID string
+
+	if len(args) != 0 && args[0] != "" {
+		queryID = args[0]
+	} else {
+		return errors.Wrap(
+			errors.New(api.LQLDeleteBadInputMsg),
```

Comment:
This is a big no-no for me. Using a defined message from the API in the CLI, as I mentioned before,
this kind of error, log, user-facing messages defined as constant could end up biting us in the but.

I'm ok if you define messages within the same Go package, but reusing messages between packages...

Let's not do this. Can we please make all constants private, please? (just define them with the first
letter as lowercase, not uppercase)

Reply:
Sure...

Reply:
I will simply test for `NotNil` instead of error msg

Reply:
Fixed.